src/hooks/use-proposals.ts

```diff
@@ -52,5 +54,12 @@ export function useProposals() {
     }
   }, [statusFilter, allProposals]);
 
-  return {proposals, setStatusFilter, statusFilter};
+  return {
+    proposals: proposals.filter(a =>
+      a.title.toLocaleLowerCase().includes(searchText.toLocaleLowerCase()),
```

Comment:
IMHO In a better way will be keep these logic on Screen, because it used only once